### PR TITLE
feat: parser for man pages

### PR DESCRIPTION
**Pull request by ribru17**
Best-effort parser for man pages, based on some common standards and such

### CODE REVIEW

File: queries/man/injections.scm [hunk 1]
```diff
@@ -0,0 +1,13 @@
+(section_heading
+  (section_title) @_title
+  (block) @injection.content
+  (#any-of? @_title "SYNOPSIS" "SYNTAX" "SINTASSI" "SKŁADNIA" "СИНТАКСИС" "書式")
+  ; a heuristic to determine if we should inject c
+  (#lua-match? @injection.content "^%s+#include")
```
Comment (clason) [inline queries/man/injections.scm:6]: I don't like this. Can the parser not do this more rigorously?

File: queries/man/injections.scm [hunk 2]
```diff
@@ -0,0 +1,13 @@
+(section_heading
+  (section_title) @_title
+  (block) @injection.content
+  (#any-of? @_title "SYNOPSIS" "SYNTAX" "SINTASSI" "SKŁADNIA" "СИНТАКСИС" "書式")
```
Comment (clason) [inline queries/man/injections.scm:4]: That's a bit of an arbitrary selection of languages. I'd prefer to stick to `en_US` only here.

**Comment by clason** [comment]
Good idea, though!

File: queries/man/injections.scm (hunk 2 above)
Comment (ObserverOfTime) [reply queries/man/injections.scm:4]: This will also include any "Feature Test Macro Requirements" lines.

**Comment by clason** [comment]
Thinking more about this, I'm a bit lukewarm. Not _everything_ needs a tree-sitter parser, and for a simple static man page Vim's simple regex syntax file seems a better tool for the job. (Especially if you need hackery to achieve feature parity.)

**Comment by lucario387** [comment]
And it's still not feature parity :)
Because who said manpages can only have C code

**Comment by ObserverOfTime** [comment]
I do think a man parser would be nice, but it would require a lot more development.
You'll know it's ready when manpages like `BASH(1)` and `printf(3)` look better than they do now.

**Comment by clason** [comment]
> And it's still not feature parity :) Because who said manpages can only have C code

Bram. Because the `man` syntax file only considers C, too. (But that would indeed be something we could do better and worth doing.)

**Comment by lucario387** [comment]
A man parser should be parsing the `man[1-9]/*` files, not the result of parsing that file though, no? :thinking:

**Comment by clason** [comment]
Yes, but how is highlighting only C _not_ feature parity, if the syntax file doesn't do any more? For me, that is the definition of "parity".

**Comment by ObserverOfTime** [comment]
> A man parser should be parsing the `man[1-9]/*` files, not the result of parsing that file though, no? 🤔

Technically that would be a (n)roff parser.

**Comment by clason** [comment]
Anyway, `man` (output) is an even worse format than vimdoc. I don't think it makes much sense putting lipstick on _that_ pig. A better use of time would be improving the `man` plugin to do better rendering (with an `(n)roff` tree-sitter parser, natch). But if you go that route, you should just render as markdown and use _that_ parser, while you're at it.

(Neovim already has a markdown renderer for LSP hover, signature help, and (soon) diagnostics.)

**Comment by ribru17** [comment]
Sounds good, I will close this. I think it is too difficult to get proper parity without the Markdown solution anyway: there are some sections that are not really syntactically different from regular text sections, yet they require C injections and the regular syntax file knows how to do it somehow. I don't know what magic is going on under the hood but it seems to be too tall of an order for this parser lol

**Pull request closed without merging**